typescript/src/entities/product.test.ts

```diff
@@ -96,3 +143,50 @@ test('can list products from server with category', () => {
     expect((d[0].categories as any)[0].name).toBe('c1');
   });
 });
+
+test('can save product', () => {
+  const merchi = new Merchi();
+  const c1 = new merchi.Category();
+  const p = new merchi.Product();
+  const c2 = new merchi.Category();
+  const d = new merchi.Domain();
+  p.categories = [c2];
+  p.domain = d;
+  c1.products = [p];
+  c1.save();
+  d.domain = "3onrb6o4";
+  p.name = "pHyz7ZucK#";
+  c2.name = "8&OaUsDgJ$ev3FYZ3";
+  p.save();
+  const fetch = mockFetch(true, {}, 200);
+  c1.save();
+  const correct =  [['products-0-name', 'pHyz7ZucK#'],
```

Comment:
I personally think make correct  =  p.toFormData() due avoid duplicate testing

as p.toFormData has been tested in another testcase

Reply:
I will approve it and leave this as suggestion

Reply:
that's not a bad idea. i'll think about it for next time